artifact-manifest/artifact-manifest.md

```diff
@@ -0,0 +1,462 @@
+    "size": 7097,
+    "digest": "sha256:c8562eaf9d81c779cbfc318d6e01b8e6f86907f1d41233268a2ed83b2f34e748"
+  },
+  "layers": [
+    {
+      "mediaType": "application/vnd.oci.image.layer.v1.tar+gzip",
+      "size": 27108069,
+      "digest": "sha256:a076a628af6f7dcabc536bee373c0d9b48d9f0516788e64080c4e841746e6ce6"
+    },
+    {
+      "mediaType": "application/vnd.oci.image.layer.v1.tar+gzip",
+      "size": 1741,
+      "digest": "sha256:f6c208f3f991dcbc417fed8efde391f887c0551d77ed0c1a125fd28f4841e1cc"
+    }
+  ]
+}
+```
+
+### Container Image, with Signatures
+
+![mysql image copy](./media/mysql-with-sigs-copy.svg)
+
+In this scenario, Notary v2 signatures have been added. Note the directionality of the Notary v2 signatures references. The `mysql:8` image has no reference to the signatures. The signatures may be added to existing artifacts. The signatures declare a reference `-->` to the `mysql:8` image. This assure the `mysql:8` image tag and digest need not change to support signatures added throughout the secured supply chain.
+
+From a user experience perspective, copying a container from a public registry to a private registry would likely be expected to copy the signatures alongside the artifact they've signed.
+
+The `oci.artifact.manifest` supports the Notary v2 requirements, including:
+
+- support for additive signatures, assuring the target manifest digest and tag do not change
+- support for multiple signatures. In the above scenario, `mysql` signed the original image. As `mysql` was copied to Docker Hub, an additional `docker community signature` was added, providing certified content attestation. Once the image copy to ACME Rockets is completed, an additional `acme-rockets signature` is added providing assurance the `mysql:8` image was security scanned and verified it's applicable to the ACME Rockets environment.
+
+From a user experience, the signature artifacts have no unique value beyond the artifact they represent, therefore they would be persisted to a registry in a form by which they are known to copy with the artifact, and be deleted when the artifact they are associated with is deleted.
+
+To support hard references, an additional dependencies collection is added to a new `application/vnd.oci.artifact.manifest.v1+json` schema. The `dependencies` collection declares the artifact the signature is dependent upon. Similar to pypi packages, the validation of dependencies are deferred. If the dependent artifacts are found, they are valid. Depending on the validation options, a signature may exist without its dependent artifact.
+
+**A Notary v2 signature of the `mysql:8` image example:**
+
+```json
+{
+  "mediaType": "application/vnd.oci.artifact.manifest.v1+json",
+  "artifactType": "application/vnd.cncf.notary.v2",
+  "config": {
+    "mediaType": "application/vnd.cncf.notary.config.v2",
+    "digest": "sha256:b5b2b2c507a0944348e0303114d8d93aaaa081732b86451d9bce1f432a537bc7",
+    "size": 102
+  },
+  "blobs": [
```

Comment:
Suppose Notary signatures were used to sign the base OS of `mysql:8`, I would imagine the blob and dependency list would be appended to in building the final `mysql:8` image. Do you think so? In which case, how would a client know which signature is for which image?

Reply:
We haven't targeted layer signing yet.